### ISSUE TITLE
[LogRhythm] Create the connector (feed IOCs, stream)

### ISSUE THREAD

**Issue by trappersec** [OpenCTI-Platform/connectors#434]
Hi,

I would like to get some guidance of how exactly I can pass on the feeds from the OpenCTI platform (not sure if relevant but runs in Docker) into SIEM, namely LR.
Looking through the previous issues I got some rough idea (Redis, GraphQL API) but I was looking through the docs and code and I am still unsure on how exactly this could be done. Any help appreciated.

(Migrated from : https://github.com/OpenCTI-Platform/docker/issues/35)

**Comment by nor3th** [comment]
Hey

I am still not sure of how MISP does the information ingestion into LogRythm. I only found this StackExchange thread (https://security.stackexchange.com/questions/230431/how-can-i-add-a-misp-stix-feed-to-logrhythm) , where the solution seems to be a custom script which interacts with the API. This would be theoretically also possible with OpenCTI. What data do you want to migrate to LogRythm?

While it is true that there are by now a couple of connectors for other SIEMs, those connectors usually pull data from the SIEM and ingest it into OpenCTI. What might be an even better way to do, is to leave OpenCTI as the "single source of truth" and just do lookups from LogRythm on the OpenCTI DB.

**Comment by trappersec** [comment]
Hi,

In case of MISP, LR can ingest TAXI feed as per the documentation : 
https://docs.logrhythm.com/docs/enterprise/other-logrhythm-applications/threat-intelligence-service/threat-intelligence-service-user-guide/add-a-custom-stix-taxii-provider/add-a-stix-feed-in-tis

Typical use case : MISP collects IOCs from various feeds, these IOCs are pushed to LR into rules to see if the IOCs has been observed in the environments monitored

My idea was to sort of "replace" MISP in this function as OpenCTI seems to have additional feeds (or at least the ability to do so) and since I am rebuilding feeds from scratch, I wanted to opt for OpenCTI as I prefer it over MISP. 

I want to get all of the MISP feeds + OpenCTI feeds into OpenCTI and then push all of the IOCs (from cases and feeds) into LR to correlate with what we see in the logs.

In case of LR and MISP it is really simple, you just specify the TAXI server URL (at least from what I can see) and that's it. Then you can integrate the IOCs into the appropriate rules in LR.

Yes, if there were a connector that would be amazing. But I understand that this takes time. So I am looking for another way to substitute this "connector" functionality. 

In simple terms, I am looking for a URL that I can put in LR which would then pull all of the IOCs from what's been ingested in OpenCTI.

Please let me know if you need more details. I am pretty sure the solution is not super difficult, it's just that I have never set anything up like this before. 

Thank you very much for your time.

**Comment by nor3th** [comment]
If I understand you correctly, then you are for now mostly interested in pulling only the Indicators from OpenCTI? 

Most connectors so far pull information from somewhere else into OpenCTI. It'd be possible to create a TAXII server connector, but that will take a bit to implement.

A quick an dirty solution might be to interact with the OpenCTI API like this https://github.com/OpenCTI-Platform/client-python/blob/master/examples/get_all_indicators_using_pagination.py . The output is then either stored as a file and ingested into LR or the same script then pushes the data into LR directly. (https://logrhythm.com/6-steps-to-using-the-logrhythm-api/)
There's an unfinished example `export_async_of_indicators.py` which I think would match your use case?

**Comment by Jipegien** [comment]
@SamuelHassine Is this issue solved with our Taxii collections?

**Comment by SamuelHassine** [comment]
Connector has been released in 5.10.
